adafruit_ble/advertising/__init__.py

```diff
@@ -0,0 +1,287 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2018 Dan Halbert for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+Advertising is the first phase of BLE where devices can broadcast
+"""
+
+import struct
+import gc
+
+def to_hex(b):
+    """Pretty prints a byte sequence as hex values."""
+    # pylint: disable=invalid-name
+    return " ".join(["{:02x}".format(v) for v in b])
+
+def to_bytes_literal(b):
+    """Prints a byte sequence as a Python bytes literal that only uses hex encoding."""
+    # pylint: disable=invalid-name
+    return "b\"" + "".join(["\\x{:02x}".format(v) for v in b]) + "\""
+
+def decode_data(data, *, key_encoding="B"):
+    """Helper which decodes length encoded structures into a dictionary with the given key
+       encoding."""
+    i = 0
+    data_dict = {}
+    if len(data) > 255:
+        print("original", data)
+        raise RuntimeError()
```

Comment:
Can this more more descriptive? `RuntimeError("Broadcast data exceeded maximum length of 255")`?

Reply:
Removed. I had it in for debugging.